File: Leetcode/PY/Leetcode495-提莫攻击.py
from typing import List
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        res,last = 0,-1
        for x in timeSeries:
            t = x + duration - 1
            res += duration if last < x else t - last
            last = t
        return res

File: Leetcode/sbgo/Leetcode495-提莫攻击.go
package sbgo

func findPoisonedDuration(timeSeries []int, duration int) int {
	ans, last := 0, -1
	for _, x := range timeSeries {
		t := x + duration - 1
		if x > last {
			ans += duration
		} else {
			ans += t - last
		}
		last = t
	}
	return ans
}
